app/models/user.rb

```diff
@@ -53,7 +53,8 @@ class User < ActiveRecord::Base
 
   scope :admin, -> { where(is_admin: true) }
 
-  validates :email, :password, :password_confirmation, presence: true
+  validates :email, presence: true
+  validates :password, :password_confirmation, presence: true, on: :create
```

Comment:
If you remove on: :create, the profile/edit page will not work. Because they both operate on the same User model.